lisa/lisa-analyses/src/main/java/it/unive/lisa/analysis/string/Bricks.java

```diff
@@ -0,0 +1,414 @@
+package it.unive.lisa.analysis.string;
+
+import it.unive.lisa.analysis.Lattice;
+import it.unive.lisa.analysis.SemanticDomain;
+import it.unive.lisa.analysis.SemanticDomain.Satisfiability;
+import it.unive.lisa.analysis.SemanticException;
+import it.unive.lisa.analysis.nonrelational.value.BaseNonRelationalValueDomain;
+import it.unive.lisa.analysis.representation.DomainRepresentation;
+import it.unive.lisa.analysis.representation.StringRepresentation;
+import it.unive.lisa.program.cfg.ProgramPoint;
+import it.unive.lisa.symbolic.value.Constant;
+import it.unive.lisa.symbolic.value.operator.binary.*;
+import it.unive.lisa.util.numeric.IntInterval;
+import it.unive.lisa.util.numeric.MathNumber;
+import java.util.*;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * The bricks string abstract domain.
+ *
+ * @author <a href="mailto:vincenzo.arceri@unipr.it">Vincenzo Arceri</a>
+ * @author <a href="mailto:sergiosalvatore.evola@studenti.unipr.it">Sergio
+ *             Salvatore Evola</a>
+ *
+ * @see <a href=
+ *          "https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34">
+ *          https://link.springer.com/chapter/10.1007/978-3-642-24559-6_34</a>
+ */
+public class Bricks implements BaseNonRelationalValueDomain<Bricks> {
+
+	private final List<Brick> bricks;
+
+	private final static Bricks TOP = new Bricks();
+
+	private final static Bricks BOTTOM = new Bricks(new ArrayList<>());
+
+	private final static int kL = 10;
```

Comment:
We should make these constants public and not final, so that we can change them before executing the analysis if we want to tune the widening.

Reply:
You mean the three constants kL, kS, and kI, correct?

Reply:
If I try to change those constants to public and not final, then the test `testDomainsAndLattice` from the class `EqualityContractVerificationTest` fails.

The error is `java.lang.AssertionError: EqualsVerifier found a problem in class it.unive.lisa.analysis.string.Bricks.
-> Significant fields: equals does not use kL, or it is stateless.` 

Then when updating the equals method, the error becomes:

`java.lang.AssertionError: EqualsVerifier found a problem in class it.unive.lisa.analysis.string.Bricks.
-> Mutability: equals depends on mutable field kL.`